markets/loggers/loggers.go

```diff
@@ -27,3 +28,11 @@ func RetrievalClientLogger(event retrievalmarket.ClientEvent, deal retrievalmark
 func RetrievalProviderLogger(event retrievalmarket.ProviderEvent, deal retrievalmarket.ProviderDealState) {
 	log.Infow("retrieval event", "name", retrievalmarket.ProviderEvents[event], "deal ID", deal.ID, "receiver", deal.Receiver, "state", retrievalmarket.DealStatuses[deal.Status], "message", deal.Message)
 }
+
+type RetrievalEvent struct {
+	Event         retrievalmarket.ClientEvent
+	Status        retrievalmarket.DealStatus
+	BytesReceived uint64
+	FundsSpent    abi.TokenAmount
+	Err           error
```

Comment:
I don't think the error type will round trip over the json rpc api very well. You might have to flatten this into a string.